### DISCUSSION TITLE
how to disable diagnostics entirely?

### DISCUSSION THREAD

**Discussion by fommil** [joaotavora/eglot#1325] [Q&A]
Hi all,

I don't like live diagnostics for a variety of reasons, so I like to turn them off. I can visually disable them with

```
(add-to-list 'eglot-stay-out-of 'flymake)
```

but I still see a bunch of things in my minibuffer when I save a file, implying that the diagnostics are still being calculated! e.g. in `rust-mode`

```
[server-notification] Fri Dec  1 15:02:08 2023:
(:jsonrpc "2.0" :method "$/progress" :params
          (:token "rust-analyzer/flycheck/0" :value
                  (:kind "end")))
```

There doesn't appear to be an option to disable diagnostics in the `eglot-ignored-server-capabilities` list.

How can I disable the calculation of diagnostics?

**Comment by joaotavora** [comment]
Eglot can't control by itself when the `rust-analyzer` server server decides do send a progress notification.  So if for some reason you don't want these progress notificaitons, you must configure the rust analyzer server itself by looking up its manual.

**Reply by fommil** [reply]
argh, I actually knew that already... I've literally written diagnostics for scala and haskell LSPs and completely forgot about this. Thanks for the reminder. I actually don't think I even gave the user the option to disable those things (although my servers were just relaying the batch compiler so were not doing any work), but I hope the `rust-analyzer` does.

**Reply by fommil** [reply]
closing the loop
https://users.rust-lang.org/t/disable-diagnostics-in-rust-analyzer/103323
